internal/gitea/repo.go

```diff
@@ -0,0 +1,190 @@
+package gitea
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"code.gitea.io/sdk/gitea"
+	"github.com/gomarkdown/markdown"
+	"github.com/jnsgruk/releasegen/internal/repos"
+)
+
+const giteaReleasesPerRepo = 3
+
+// Repository represents a single repository. Note that this might be a Gitea
+// Repository or it might be a folder in a Gitea Repository if the Repository
+// is a monorepo.
+type Repository struct {
+	Details       repos.RepoDetails
+	org           string // The Gitea Org that owns the repo.
+	client        *gitea.Client
+	defaultBranch string
+	folder        string
+}
+
+// Process populates the Repository with details of its releases, and commits.
+func (r *Repository) Process() error {
+	// Iterate over the releases in the Gitea repo and add them to our repository's details.
+	err := r.processReleases()
+	if err != nil {
+		return err
+	}
+
+	if len(r.Details.Releases) > 0 {
+		// Calculate the number of commits since the latest release.
+		err := r.processCommitsSinceRelease()
+		if err != nil {
+			return err
+		}
+	} else {
+		// If there are no releases, get the latest commit instead.
+		err := r.processCommits()
+		if err != nil {
+			return err
+		}
+	}
+
+	// Populate the repository's README from Gitea, parse any linked snaps or charms.
+	err = r.parseReadme()
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// parseReadme is a helper function to fetch the README from a Gitea repository and return
+// its contents as a string.
+func (r *Repository) parseReadme() error {
+	var readmeNames = []string {"README.md", "README.rst"}
```

Comment:
Typical Go style would be to use `:=` in this case:
```suggestion
	readmeNames := []string{"README.md", "README.rst"}
```

Also, I notice this code hasn't been `go fmt`ed. You probably want to run `go fmt ./...` over all the code before committing. Also, that should probably be a CI test. I usually do something like [this](https://github.com/canonical/pebble/blob/91bba6711035c849b7462e883f1109fe16c1217d/.github/workflows/tests.yml#L66-L69)

Naming nit: this function is `parseReadme` so probably don't need the `readme` or `r` prefix on these variable names. Maybe just `names` or `filenames`.